### ISSUE TITLE
Tests fail when all packages are loaded

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/liepring#9]
This happens because of the clash with `SC` defined as an operation in the SimpComp package:

https://github.com/simpcomp-team/simpcomp/blob/8b8c1ed52e9cbc1a73dab161579654344c99396e/lib/generate.gi#L257-L283

The first diff is the actual failure, the rest is caused by a domino effect.

Renaming `SC` to `sc` will likely help.

```
testing: /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-quicktest/\
GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-update-sta\
ble-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:4
# Input is:
SC := rec( dim := 3, prime := 2, tab := [] );;
# Expected output:
# But found:
Error, Variable: 'SC' is read only
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:5
# Input is:
L := LiePRingBySCTable(SC);
# Expected output:
<LiePRing of dimension 3 over prime 2>
# But found:
Error, Record IsBound: <rec> must be a record (not a function)
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:7
# Input is:
l := BasisOfLiePRing(L);
# Expected output:
[ l1, l2, l3 ]
# But found:
Error, Variable: 'L' must have a value
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:9
# Input is:
l[1]*l[2];
# Expected output:
0
# But found:
Error, Variable: 'l' must have a value
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:11
# Input is:
2*l[1];
# Expected output:
0
# But found:
Error, Variable: 'l' must have a value
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:13
# Input is:
l[1] + l[2];
# Expected output:
l1 + l2
# But found:
Error, Variable: 'l' must have a value
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:15
# Input is:
SC := rec( dim := 4, prime := 5, tab := [ [], [3, 1], [], [4, 1]]);;
# Expected output:
# But found:
Error, Variable: 'SC' is read only
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:16
# Input is:
L := LiePRingBySCTableNC(SC);;
# Expected output:
# But found:
Error, Record IsBound: <rec> must be a record (not a function)
########
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-stable-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/packages/label/kovacs/GAP-pkg-u\
pdate-stable-snapshot/pkg/liepring-1.9.1/tst/manual.example-3.tst:17
# Input is:
ViewPCPresentation(L);
# Expected output:
[l2,l1] = l3
[l3,l1] = l4
# But found:
Error, Variable: 'L' must have a value
########
```

**Comment by olexandr-konovalov** [comment]
BTW, https://www.gap-system.org/Manuals/doc/ref/chap76.html#X7DEACD9786DE29F1 says:

> Additionally, there is a recommended naming convention that the GAP core system and GAP packages should not use global variables starting in the lowercase. This allows to reserve variables with names starting in lowercase to the GAP user so they will never clash with the system. It is extremely important to avoid using for package global variables very short names started in lowercase. For example, such names like cs, exp, ngens, pc, pow which are perfectly fine for local variables, should never be used for globals. Additionally, the package must not have writable global variables with very short names even if they are starting in uppercase, for example, C1 or ORB, since they also could be easily overwritten by the user.

so I'd argue that if tests use variables in starting in lowercase, then they will not clash with any packages. Cf. also the PR https://github.com/gap-system/gap/pull/1633 by @ChrisJefferson which allows local variables in test files - that would be another way to fix this test, but that PR is yet to be merged. 

I'd also suggest that SimpComp should avoid using `SC`, but that ship had sailed already.

@fingolfin to submit a fix, shall I change `tst/manual.example-3.tst`, or find the location of the example in the source for the manual (or both)?

**Comment by olexandr-konovalov** [comment]
The problem can be seen now at https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.10-staging/jobs/439531180

**Comment by fingolfin** [comment]
Resolved in version 1.9.2
